mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/ILeaderMonitorFactory.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.core;
+
+import io.mantisrx.server.core.master.MasterMonitor;
+
+public interface ILeaderMonitorFactory {
```

Comment:
I separated the interface for monitor and leader elector to match the split usage.  Logically this did make sense to me, but I'm open to adjusting this.